modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/FormBuilder/withActionableFields.es.js

```diff
@@ -293,7 +292,7 @@ const withActionableFields = (ChildComponent) => {
 		}
 
 		_hasLeftField(relatedTarget) {
-			return !relatedTarget.closest(
+			return !relatedTarget?.closest(
```

Comment:
Should we cover for misuse? :thinking: 
```
_hasLeftField(null) // true
```
This might hide bugs. Maybe it's better to let it burn? @wincent 

Few more cases of this below.

Reply:
@markocikos without the conditional operator, our ci tests were already failing....

The util from metal-dom also allows for falsey values and returns them.

```js
dom.closest(null, '.some-selector') === null
dom.closest(false, '.some-selector') === false
dom.closest(0, '.some-selector') === 0
```

Because metal-dom did this, I am tempted to add this anytime we use `closest`

Reply:
> our ci tests were already failing....

Maybe this was because of the Browser in CI as well?

Reply:
@jbalsas I was able to reproduce it locally as well, so I don't think it was browser based.

In a lot of these cases where we are working with nodes, its really hard to know what to expect on if the node should exist or not. A lot of the metal-dom utils allow for `null` when passing an element through which means we likely have a lot of cases where the element may not exist.

Reply:
Got it! Thought maybe `?` was not supported there yet...

Reply:
Actually I think it was the unit tests that were failing for my fix.

I'm sure poshi will have issues with it though.


Gonna wait to re-test until browsers are updated

Reply:
> In a lot of these cases where we are working with nodes, its really hard to know what to expect on if the node should exist or not. A lot of the metal-dom utils allow for null when passing an element through which means we likely have a lot of cases where the element may not exist.

Well, it's not misuse if it's intended use :sweat_smile:  It's not a good practice, in my opinion, but updating usages not to break is probably beyond the scope of this PR. If things do break, it's probably best to just keep the optional chaining.

Reply:
Good discussion. What we have here are two competing values which are independently good but hard to combine:

- Don't swallow errors; you might be masking a bug.
- Preserve the behavior of an existing API.

Now, the old API was definitely questionable (I mean, `dom.closest(0, '.some-selector') === 0`...), but keeping the same shape does seem to be the path of least risk. And like @bryceosterhaus says below, the optional chaining is a nice-to-have, but that's all it is.